### PR TITLE
refactor: Table to CSS-IN-JS

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
还有一个问题，筛选菜单本地和线上样式完全不一样

- http://127.0.0.1:8001/components/table-cn/#components-table-demo-head
- https://preview-35584-ant-design.surge.sh/components/table-cn/#components-table-demo-head

截屏2022-05-29 下午7 32 00

不知道本地表格筛选菜单为啥会被 Menu 的样式污染，然而本地的 Dropdown 演示又没有问题。

**Comment by zombieJ** [comment]
哎，和 Dropdown 当时遇到的同一个问题：https://github.com/react-component/dropdown/blob/master/src/Dropdown.tsx#L122

`rc-dropdown` 里 clone 了 element 加属性。Table 这边没有直接包 Menu 所以导致跳过 Menu 样式注入的代码就失效了。

**Comment by afc163** [comment]
> rc-dropdown 里 clone 了 element 加属性。Table 这边没有直接包 Menu 所以导致跳过 Menu 样式注入的代码就失效了。

有啥解法？

**Comment by zombieJ** [comment]
> > rc-dropdown 里 clone 了 element 加属性。Table 这边没有直接包 Menu 所以导致跳过 Menu 样式注入的代码就失效了。
> 
> 有啥解法？

rc-dropdown 里的注入属性拆掉，onClick 部分移至 rc-trigger。其余部分通过 Context 在 antd 侧传递。

File: components/table/style/expand.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  // FIXME: 需要从 checkbox 那里取
+  const checkboxSize = token.controlInteractiveSize;
+  const halfInnerSize = checkboxSize / 2 - 1;
+  // must be odd number, unless it cannot align centerly
+  const expandIconSize = halfInnerSize * 2 + token.controlLineWidth * 3;
+  const tableBorder = `${token.controlLineWidth}px ${token.controlLineType} ${token.tableBorderColor}`;
+  return {
+    [`${componentCls}-wrapper`]: {
+      [`${componentCls}-expand-icon-col`]: {
+        width: 48,
```
Comment (afc163) [inline components/table/style/expand.tsx:17]:
```suggestion
        // FIXME
        width: 48,
```

File: components/table/style/filter.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          },
+        },
+      },
+    },
+    {
+      // Dropdown
+      [`${antCls}-dropdown`]: {
+        [tableFilterDropdownPrefixCls]: {
+          ...resetComponent(token),
+          minWidth: 120,
```
Comment (afc163) [inline components/table/style/filter.tsx:50]:
```suggestion
          // FIXME
          minWidth: 120,
```

**Comment by zombieJ** [comment]
https://preview-35584-ant-design.surge.sh/components/table-cn/#components-table-demo-head 

应该好了

**Comment by afc163** [comment]
666

**Pull request merged**